src/libs/elektra/keymeta.c

```diff
@@ -483,19 +482,7 @@ ssize_t keySetMeta (Key * key, const char * metaName, const char * newMetaString
 	if (newMetaString != NULL)
 	{
 		/*Add the meta information to the key*/
-		metaStringDup = elektraMemDup (newMetaString, metaStringSize);
-		if (metaStringDup == NULL)
-		{
-			// TODO: actually we might already have changed
-			// the key
-			keyDel (toSet);
-			return -1;
-		}
-
-		if (toSet->data.v && !test_bit (toSet->flags, KEY_FLAG_MMAP_DATA)) elektraFree (toSet->data.v);
-		clear_bit (toSet->flags, (keyflag_t) KEY_FLAG_MMAP_DATA);
-		toSet->data.c = metaStringDup;
-		toSet->dataSize = metaStringSize;
+		keySetRaw (toSet, newMetaString, metaStringSize);
```

Comment:
Is there any reason why `keySetMeta` duplicated parts of the functionality of `keySetRaw` here?

Reply:
Nice catch! :heart: I don't think so.

Reply:
I guess it was because of the `TODO`, or maybe this was simply implemented before `keySetRaw` existed.